segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaConnectionListener.java

```diff
@@ -105,7 +110,17 @@
     @VisibleForTesting
     public PravegaConnectionListener(boolean enableTls, int port, StreamSegmentStore streamSegmentStore, TableStore tableStore) {
         this(enableTls, false, "localhost", port, streamSegmentStore, tableStore, SegmentStatsRecorder.noOp(), TableSegmentStatsRecorder.noOp(),
-                new PassingTokenVerifier(), null, null, true);
+                new PassingTokenVerifier(), null, null, true,
+                ExecutorServiceHelpers.newScheduledThreadPool(1, "test-token-expiry-handler"));
+    }
+
+    public PravegaConnectionListener(boolean enableTls, boolean enableTlsReload, String host, int port, StreamSegmentStore streamSegmentStore, TableStore tableStore,
+                                     SegmentStatsRecorder statsRecorder, TableSegmentStatsRecorder tableStatsRecorder,
+                                     DelegationTokenVerifier tokenVerifier, String certFile, String keyFile,
+                                     boolean replyWithStackTraceOnError) {
+        this(enableTls, enableTlsReload, host, port, streamSegmentStore, tableStore, statsRecorder, tableStatsRecorder,
+                tokenVerifier, certFile, keyFile, replyWithStackTraceOnError,
+                ExecutorServiceHelpers.newScheduledThreadPool(1, "test-token-expiry-handler"));
```

Comment:
In this code path, this executor does not look like it will ever be shutdown. That is a thread leak.

Reply:
That constructor was being used in unit tests. I've modified it to inject the executor from outside, and have modified the tests.